Source/Scene/Model.js

```diff
@@ -2533,6 +2707,85 @@ define([
         }
     }
 
+    function scaleFromMatrix5Array(matrix) {
+        return [matrix[0], matrix[1], matrix[2], matrix[3],
+                matrix[5], matrix[6], matrix[7], matrix[8],
+                matrix[10], matrix[11], matrix[12], matrix[13],
+                matrix[15], matrix[16], matrix[17], matrix[18]];
+    }
+
+    function translateFromMatrix5Array(matrix) {
+        return [matrix[20], matrix[21], matrix[22], matrix[23]];
+    }
+
+    function createUniformsForQuantizedAttributes(model, primitive, context) {
+        var gltf = model.gltf;
+        var accessors = gltf.accessors;
+        var programId = getProgramForPrimitive(model, primitive);
+        var quantizedUniforms = model._quantizedUniforms[programId];
+        var setUniforms = [];
+        var uniformMap = {};
+
+        for (var attribute in primitive.attributes) {
+            if (primitive.attributes.hasOwnProperty(attribute)) {
+                var accessorId = primitive.attributes[attribute];
+                var a = accessors[accessorId];
+                var extensions = a.extensions;
+
+                if (defined(extensions)) {
+                    var quantizedAttributes = extensions.WEB3D_quantized_attributes;
+                    if (defined(quantizedAttributes)) {
+                        var decodeMatrix = quantizedAttributes.decodeMatrix;
+                        var uniformVariable = 'czm_u_dec_' + attribute.toLowerCase();
+
+                        switch (a.type) {
+                            case 'VEC2':
```

Comment:
The allowed types are https://github.com/KhronosGroup/glTF/tree/master/specification#parametertype-white_check_mark

I think the extension spec should limit the types - at least exclude booleans, but perhaps also exclude integers if it makes sense to do so.